### PR TITLE
Remove class inheritance hacks from plugin API.

### PR DESCRIPTION
**Pull request by jjrv**

### CODE REVIEW

File: src/plugin/AMD.ts [hunk 1]
```diff
@@ -1,21 +1,13 @@
 import { Record, ModuleFactory } from '../Record';
 import { ModuleAMD } from '../Module';
 import { globalEnv, globalEval } from '../platform';
-import { Loader, LoaderConfig } from '../LoaderBase';
+import { Loader, LoaderPlugin } from '../Loader';
 
 /** AMD (asynchronous module definition) loader plugin. */
 
-export class AMD extends Loader {
+export const AMD = (loader: Loader): LoaderPlugin => {
```
Comment (ISNIT0) [inline src/plugin/AMD.ts]: Why a function that returns a `LoaderPlugin` instead of a class that implements `LoaderPlugin`?

Comment (jjrv) [reply src/plugin/AMD.ts]: I was also considering that. Might be better. I'll prototype that option too so we can compare the approaches.

**Comment by jjrv** [comment]
Minified size grew by 779 bytes, which doesn't matter if the code is rather clearer with classes. Opinions?

**Comment by ISNIT0** [comment]
LGTM 👍

**Pull request merged**